vllm/core/block_manager_v1.py

```diff
@@ -290,21 +321,53 @@ def allocate(self, seq_group: SequenceGroup) -> None:
                     and logical_idx >= self.block_sliding_window):
                 block = block_table[logical_idx % self.block_sliding_window]
                 # Set the reference counts of the token blocks.
-                block.ref_count = seq_group.num_seqs()
-            elif self.enable_caching:
+                block.ref_count = ref_count
+            elif not is_encoder_decoder and self.enable_caching:
                 block = self.gpu_allocator.allocate(
                     seq.hash_of_block(logical_idx),
                     seq.num_hashed_tokens_of_block(logical_idx))
             else:
                 block = self.gpu_allocator.allocate()
                 # Set the reference counts of the token blocks.
-                block.ref_count = seq_group.num_seqs()
+                block.ref_count = ref_count
             block_table.append(block)
 
-        # Assign the block table for each sequence.
+        return block_table
+
+    def allocate(self, seq_group: SequenceGroup) -> None:
+        encoder_seq = seq_group.get_encoder_seq()
+        is_encoder_decoder = seq_group.is_encoder_decoder()
+
+        if (self.block_sliding_window is not None) and \
+           is_encoder_decoder:
+            raise NotImplementedError(str_not_impl_enc_dec_swa)
+
+        if self.enable_caching and is_encoder_decoder:
+            raise NotImplementedError(str_not_impl_enc_dec_prefix_cache)
```

Comment:
have two ifs within a single `if is_encoder_decoder` here? To avoid extra check in most common case

Reply:
Thanks! Fixed in most recent push.